docs/rfcs/RFC0004-Fusilli-IREE-Kernel-Provider-hipDNN.md

```diff
@@ -54,30 +54,126 @@ From a code organization standpoint, there are three components to reason about:
    It depends minimally on IREE compiler (CLI) and IREE runtime (C-API), and
    does NOT require a direct HIP dependency (abstracted by IREE's HAL design).
-1. The hipDNN engine plugin for Fusilli. This specializes Fusilli for use within
+1. Fusilli-Plugin. The hipDNN engine plugin for Fusilli. This specializes Fusilli for use within
    hipDNN specifically for AMD GPUs. Currently it is being developed
-   [here](https://github.com/nod-ai/shark-ai/tree/main/fusilli-plugin).
+   [here](https://github.com/nod-ai/shark-ai/tree/main/fusilli-plugin), and has
+   begun a migration to
+   [rocm-libraries](https://github.com/ROCm/rocm-libraries/tree/develop/projects/fusilli-plugin).
    In addition to Fusilli's dependencies, the plugin also depends on HIP, hipDNN
    frontend/SDK and hipDNN's dependencies transitively.
 
 ### Short term plan
 
-The immediate workplan is to move the hipDNN engine plugin (i.e., component 3
-above) into `rocm-libraries` (under `dnn-providers` once build tree is normalized
-per RFC0003) following guidelines from the MIOpen plugin restructuring effort.
-This will be built conditionally (NOT on by default) and will pull in Fusilli
-and IREE as external dependencies.
-
-The expected build artifact from the plugin integration is a self-contained
-`libfusilliplugin.so` that is linked against Fusilli headers and IREE runtime
-sources built and statically linked. The dependency on the IREE compiler is
-through the `iree-compile` binary (made available typically through a pip-install),
-as Fusilli currently invokes the compiler through its command-line-interface.
+The immediate goal is to build the hipDNN engine plugin (i.e., component 3
+above) in `TheRock`. The plugin's dependencies require all three components to be part of
+the build. IREE is a large dependency; the desire is to keep it an optional component
+not built by default. Therefore IREE's `TheRock` build and all dependent projects
+(Fusilli, Fusilli-Plugin) will be gated behind a CMake option
+`THEROCK_ENABLE_BUILD_IREE_LIBS`
+
+For the various build scripts, this RFC proposes a new top level directory
+`iree-libs`, gated in the top level `CMakeLists.txt`. Note: `iree-libs` name is
+subject to change, we need a name, so `iree-libs` is a placeholder for the RFC.
+
+```diff
+ add_subdirectory(comm-libs)
+ add_subdirectory(math-libs)
+ add_subdirectory(ml-libs)
++if(THEROCK_ENABLE_BUILD_IREE_LIBS)
++  add_subdirectory(iree-libs)
++endif()
+```
+
+```
+...
+├── iree-libs
+│   ├── CMakeLists.txt
+│   ├── fusilli
+│   │   └── CMakeLists.txt
+│   ├── fusilli-plugin
+│   │   └── CMakeLists.txt
+│   └── iree
+│       └── CMakeLists.txt
+├── math-libs
+...
+```
+
+The following sections detail where each dependency will live outside of
+`TheRock`, and what its build will look like inside of `TheRock`.
+
+#### `iree`
+
+IREE will remain in its Linux Foundation-governed `iree-org` repo. `TheRock`
```

Comment:
I think we want this to be a submodule -- we will be bumping it regularly and that is how we do it. The fetch_sources.py script can have flags to enable/disable it, and the required nested submodules can be initialized there.

So you would have iree-libs/iree (submodule)

Reply:
Ok, so you dont want to do the fetch sources here, but rather have iree as a submodule, and just initialize the nested flatcc submodule? If someone checks out the rock, and does `git submodule init --recursive` that will initialize all of iree submodules, i.e. IREE torch-mlir and LLVM submodule.

Reply:
If people init --recursive, they already get more than they likely bargained for. We drive people to fetch_sources.py to selectively init and fetch transitives for the scope they want. It's not the best, but there aren't fantastic options either.